detections/endpoint/windows_bidirectional_communication_via_telegram_bot.yml

```diff
@@ -0,0 +1,67 @@
+name: Windows Bidirectional Communication via Telegram Bot
+id: 86f66f44-94d9-412d-a71d-5d8ed0fef72e
+version: 1
+date: '2024-12-12'
+author: Teoderick Contreras, Splunk
+data_sources:
+- Sysmon EventID 22
+type: Anomaly
+status: production
+description: The following analytic detects the use of DNS API calls associated with Telegram, which may indicate access via a Telegram bot commonly used by malware for command and control (C2) communications. By monitoring DNS queries related to Telegram's infrastructure, the detection identifies potential attempts to establish covert communication channels between a compromised system and external malicious actors. This behavior is often observed in cyberattacks where Telegram bots are used to receive commands or exfiltrate data, making it a key indicator of suspicious or malicious activity within a network.
+kind: endpoint
+search: '`sysmon` EventCode=22  query = "api.telegram.org" 
+  | stats count min(_time) as firstTime max(_time) as lastTime  by query answer QueryResults QueryStatus process_name process_guid Computer 
+  | rename Computer as dest 
+  | `security_content_ctime(firstTime)` 
+  | `security_content_ctime(lastTime)` 
+  | `windows_bidirectional_communication_via_telegram_bot_filter`'
```

Comment:
To be confirmed but I believe The logic needs to exclude the telegram binary on windows because it communicates with the API itself. 
I don't know if there is a direct link from the web but excluding browsers is will also avoid FPs